spec/tictactoe_spec.rb

```diff
@@ -8,73 +8,57 @@ class FakeIO
   def initialize(stdin: '', stdout: [])
     @stdin = stdin
     @stdout = stdout
+    @accepted_input = [*1..9].map!(&:to_s)
   end
 
-  def puts(string)
+  def output_to_user(string)
     @stdout << string
   end
 
-  def gets
-    string_to_return = @stdin
+  def prompt_user_for_position
+    string_to_return = @accepted_input.include?(@stdin) ? @stdin.to_i - 1: nil
     @stdin = ''
     string_to_return
   end
 end
 
+
 def is_row_outputed(io:, row:)
   expect(io.stdout).to include row
 end
 
+def create_new_game(io: FakeIO.new, board: TicTacToe::Board.new(layout: Array.new(9, ' ' ) ) )
+  TicTacToe.new(io: io, board: board)
+end
+
 describe TicTacToe do
   context 'when a new game is started' do
-    context 'when the board is rendered' do
-      it 'returns a single list with 9 elements' do
-        game = TicTacToe.new(io: FakeIO.new)
-
-        expect(game.render_board.length).to eql 9
-      end
-    end
-
     context 'when the board is empty' do
       it 'sets board data to empty spaces' do
-        game = TicTacToe.new(io: FakeIO.new)
-        all_spaces_empty = game.render_board.all? { |space| space == ' ' }
-        expect(all_spaces_empty).to eql true
+        game = create_new_game
+        new_game_board = game.render_board
+        expect(new_game_board).to eql " | | \n------\n | | \n------\n | | \n"
       end
 
       it 'the first token will be "X"' do
-        io = FakeIO.new
-        game = TicTacToe.new(io: io)
-
-        game.place_token(1)
-
+        io = FakeIO.new(stdin: "1")
+        game = create_new_game(io: io)
+        game.cycle_player_turn
         current_output = io.stdout
-        expect(current_output).to include /X/
+        expect(current_output).to include /Player X/
       end
     end
   end
 
-  context 'when a token is placed on an empty board' do
-    it 'renders the board in a 3x3 after a token is placed' do
-      io = FakeIO.new
-      game = TicTacToe.new(io: io)
-
-      game.place_token(3)
-      expectedBoardOutput = [" | | \n------\nX| | \n------\n | | \n"]
-
-      expect(io.stdout).to eql expectedBoardOutput
-    end
-  end
-
   context 'when the user interacts with the board' do
-    let(:desired_position) { $desired_position =  4  }
+    let(:desired_position) { $desired_position =  "5"  }
 
     context 'when the user tries to place a token in a free space' do
       it 'prompt the user to select a space' do
         io = FakeIO.new(stdin: desired_position)
-        game = TicTacToe.new(io: io)
+        game = create_new_game(io: io)
 
-        game.begin_player_turn
+        game.cycle_player_turn
```

Comment:
I wonder if `cycle_player_turn` is the best name. In the domain of TicTacToe and other turn-based games, what kind of language do people use to describe this? Do you often hear people say _cycle_?